rfcs/0089-relax-upstreaming.md

```diff
@@ -0,0 +1,775 @@
+- BYOC: We will try to reuse the existing translation spec. In Relax, BYOC can be formalized as a pass that call external packed functions.
+- Automatic scheduling: We will try to reuse the underlying [MetaSchedule](https://github.com/apache/tvm-rfcs/blob/main/rfcs/0005-meta-schedule-autotensorir.md) infrastructure, and the automatic scheduling can be formalized as a pass in Relax.
+
+We will migrate to use the new TVMScript parser/printer proposed in the following RFCs:
+
+- Parser: https://github.com/apache/tvm-rfcs/pull/79
+- Printer: https://github.com/apache/tvm-rfcs/pull/74
+
+# 6. **Rationale and alternatives**
+This proposal brings Relax as a separate dialect to the TVM ecosystem. We propose to do so because: Relax presents a set of closely chosen co-design choices to resolve pain points we observed so far in the TVM ecosystem, namely dynamic shape, side-effect handling, and cross-IR interaction. Additionally, the current Relax mainly focuses on graph-TIR-based cross-layer optimizations and complements the passes implemented in Relay.
+
+Presenting optional modules and dialects to the ecosystem is an important way to welcome new improvements, land our technical commitment timely, continue to reinvent ourselves, and welcome new community members who have new use cases.  It also aligns with practices we see in other ML projects, such as MLIR dialects, PyTorch FX modules, etc.
+
+As in these previous cases (MLIR dialects and TorchFX), there can be some amount of perceived duplications with existing modules. Nevertheless, there are usually technical reasons for a new module, both from design consistency, time to delivery, and separation of concerns from stable modules during improvements. Additionally, there is also a majority amount of support to help continue maintaining the new module. This new community effort would help us to welcome new members to the community, and enhance the overall development of TVM.
+
+We acknowledge that the migration of default flow (that involves Relay) would benefit from more in-depth discussions on possible approaches. They are not part of this proposal and we welcome further discussions on this topic. We think however, that it is useful to provide such a context. This section provides a context on possible difficulties in evolving Relay to bring some of the features.
+
+From the development point of view, we think that the technical reasonings, community needs, and technical support around the module being present is sufficient to justify the inclusion of the optional module. We agree that we need to do further in-depth discussions on default flow evolution, and would be more than happy to continue the discussion after this RFC.
```

Comment:
Can you comment on what are the discussions you think need to happen and what are the expected outcomes from these discussions?

The current wording feels a bit vague, considering Relax as a new IR that might generate uncertainty from the contributors on where to actually contribute new features, so, if TVM is moving to Relax, new features should go on Relax or if eventually we're going to stay with Relay, new features should actually Relay, and in this case, where does Relax fit in this puzzle?